testutil/collections.go

```diff
@@ -0,0 +1,45 @@
+package testutil
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+
+	storetypes "cosmossdk.io/store/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// DiffCollectionsMigration is meant to aid in the migration from the previous store
+// solution to collections. It takes a few steps to use it:
+//  1. Write a test function that writes elements using the old store solution.
+//  2. Run the test function and copy the hash that it outputs (run it a couple
+//     of times to make sure no non-deterministic data is being used).
+//  3. Change the code to use collections and run the test function again to make
+//     sure the hash didn't change.
+func DiffCollectionsMigration(
+	ctx sdk.Context,
+	storeKey *storetypes.KVStoreKey,
+	iterations int,
+	writeElem func(int64),
+	targetHash string,
+) error {
+	for i := int64(0); i < int64(iterations); i++ {
+		writeElem(i)
+	}
+
+	allkvs := []byte{}
```

Comment:
We can skip over creating a massive slice that'll be iterated upon then appended to by simply creating a sha256 hash whose .Write method can be invoked

```go
h := sha256.New()

for ...; it.Valid(); it.Next() {
   h.Write(it.Key())
   h.Write(it.Value())
}

if g, w := fmt.Sprintf("%x", h.Sum(nil)), wantHash; g != w {
```

Reply:
awesome, ty!